In [ ]:
# 구글 드라이브 Mount
from google.colab import drive
drive.mount('/content/drive/')

Mounted at /content/drive/


In [ ]:
path = '/content/drive/MyDrive/AKI/mimic'

## icu 폴더 데이터셋 로드

In [ ]:
import pandas as pd

# 파일 경로 변수 설정
caregiver_path = '/content/drive/MyDrive/AKI/mimic/3.0/icu/caregiver.csv.gz'
chartevents_path = '/content/drive/MyDrive/AKI/mimic/3.0/icu/chartevents.csv.gz'
datetimeevents_path = '/content/drive/MyDrive/AKI/mimic/3.0/icu/datetimeevents.csv.gz'
d_items_path = '/content/drive/MyDrive/AKI/mimic/3.0/icu/d_items.csv.gz'
ingredientevents_path = '/content/drive/MyDrive/AKI/mimic/3.0/icu/ingredientevents.csv.gz'
icustays_path = '/content/drive/MyDrive/AKI/mimic/3.0/icu/icustays.csv.gz'
inputevents_path = '/content/drive/MyDrive/AKI/mimic/3.0/icu/inputevents.csv.gz'
outputevents_path = '/content/drive/MyDrive/AKI/mimic/3.0/icu/outputevents.csv.gz'
procedureevents_path = '/content/drive/MyDrive/AKI/mimic/3.0/icu/procedureevents.csv.gz'

# 청크 사이즈 설정
chunk_size = 100000

# 각 파일에 대해 청크 단위로 데이터를 로드하고 첫 번째 청크만 확인하는 함수
def load_and_check_first_chunk(file_path):
    try:
        for chunk in pd.read_csv(file_path, chunksize=chunk_size, compression='gzip'):
            print(f"파일명: {file_path}")
            print(f"컬럼명: {chunk.columns.tolist()}")
            print(f"데이터 샘플:\n{chunk.head()}\n")
            break  # 첫 번째 청크만 확인
    except Exception as e:
        print(f"파일 '{file_path}'를 로드하는 도중 에러 발생: {e}")

### 1. caregiver

In [ ]:
# 중환자실에서 환자에게 처치를 수행한 의료진의 정보입니다. 각 의료진의 ID, 직위, 역할 등이 포함됩니다.

load_and_check_first_chunk(caregiver_path)

파일명: /content/drive/MyDrive/AKI/mimic/3.0/icu/caregiver.csv.gz
컬럼명: ['caregiver_id']
데이터 샘플:
   caregiver_id
0             3
1             4
2             6
3             9
4            12



In [ ]:
caregiver = pd.read_csv(caregiver_path, compression='gzip')
caregiver.drop_duplicates(inplace=True)
caregiver.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17984 entries, 0 to 17983
Data columns (total 1 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   caregiver_id  17984 non-null  int64
dtypes: int64(1)
memory usage: 140.6 KB


In [ ]:
# 결측치 확인
caregiver_missing = caregiver.isnull().sum()
print("caregiver 데이터 결측치 확인:")
print(caregiver_missing[caregiver_missing > 0])

# 중복 데이터 확인
caregiver_duplicates = caregiver.duplicated().sum()
print(f"\ncaregiver 데이터 중복된 행의 수: {caregiver_duplicates}")

# 중복 데이터가 있는 경우 확인
if caregiver_duplicates > 0:
    print("\n중복된 데이터 예시:")
    print(caregiver[caregiver.duplicated()].head())

caregiver 데이터 결측치 확인:
Series([], dtype: int64)

caregiver 데이터 중복된 행의 수: 0


### 2. chartevents

In [ ]:
# 환자의 바이탈 사인, 검사 결과, 약물 투여 등의 시계열 데이터를 포함하는 가장 큰 테이블입니다. 중환자실에서 주기적으로 측정된 환자의 상태 정보를 포함합니다.

load_and_check_first_chunk(chartevents_path)

파일명: /content/drive/MyDrive/AKI/mimic/3.0/icu/chartevents.csv.gz
컬럼명: ['subject_id', 'hadm_id', 'stay_id', 'caregiver_id', 'charttime', 'storetime', 'itemid', 'value', 'valuenum', 'valueuom', 'warning']
데이터 샘플:
   subject_id   hadm_id   stay_id  caregiver_id            charttime  \
0    10000032  29079034  39553978       18704.0  2180-07-23 12:36:00   
1    10000032  29079034  39553978       18704.0  2180-07-23 12:36:00   
2    10000032  29079034  39553978       18704.0  2180-07-23 12:36:00   
3    10000032  29079034  39553978       18704.0  2180-07-23 14:00:00   
4    10000032  29079034  39553978       18704.0  2180-07-23 14:00:00   

             storetime  itemid              value  valuenum valueuom  warning  
0  2180-07-23 14:45:00  226512               39.4      39.4       kg      0.0  
1  2180-07-23 14:45:00  226707                 60      60.0     Inch      0.0  
2  2180-07-23 14:45:00  226730                152     152.0       cm      0.0  
3  2180-07-23 14:18:00  220048  SR (

In [ ]:
chartevents = pd.read_csv(chartevents_path, compression='gzip', nrows=10000)  # 너무 큰 경우 일부만 로드

# 결측치 확인
chartevents_missing = chartevents.isnull().sum()
print("chartevents 데이터 결측치 확인:")
print(chartevents_missing[chartevents_missing > 0])

# 중복 데이터 확인
chartevents_duplicates = chartevents.duplicated().sum()
print(f"\nchartevents 데이터 중복된 행의 수: {chartevents_duplicates}")

# 중복 데이터가 있는 경우 확인
if chartevents_duplicates > 0:
    print("\n중복된 데이터 예시:")
    print(chartevents[chartevents.duplicated()].head())

chartevents 데이터 결측치 확인:
caregiver_id     241
storetime         23
value            394
valuenum        6146
valueuom        7956
warning           23
dtype: int64

chartevents 데이터 중복된 행의 수: 0


### 3. datatimeevents

In [ ]:
# 중환자실 내에서 시간 관련 이벤트를 기록한 데이터입니다. 환자의 검사 결과, 바이탈 사인 측정 등과 관련된 시간 이벤트가 포함될 수 있습니다.

load_and_check_first_chunk(datetimeevents_path)

파일명: /content/drive/MyDrive/AKI/mimic/3.0/icu/datetimeevents.csv.gz
컬럼명: ['subject_id', 'hadm_id', 'stay_id', 'caregiver_id', 'charttime', 'storetime', 'itemid', 'value', 'valueuom', 'warning']
데이터 샘플:
   subject_id   hadm_id   stay_id  caregiver_id            charttime  \
0    10000032  29079034  39553978         18704  2180-07-23 14:24:00   
1    10000032  29079034  39553978         18704  2180-07-23 14:24:00   
2    10000032  29079034  39553978         20925  2180-07-23 21:02:00   
3    10000032  29079034  39553978         20925  2180-07-23 21:02:00   
4    10000690  25860671  37081114         17393  2150-11-02 20:00:00   

             storetime  itemid                value valueuom  warning  
0  2180-07-23 14:24:00  225754  2180-07-23 14:24:00     Date        0  
1  2180-07-23 14:24:00  225755  2180-07-23 14:24:00     Date        0  
2  2180-07-23 21:02:00  225754  2180-07-23 00:00:00     Date        0  
3  2180-07-23 21:02:00  225755  2180-07-23 00:00:00     Date        0  
4  21

In [ ]:
datetimeevents = pd.read_csv(datetimeevents_path, compression='gzip')
datetimeevents['charttime'] = pd.to_datetime(datetimeevents['charttime'], errors='coerce')
datetimeevents['storetime'] = pd.to_datetime(datetimeevents['storetime'], errors='coerce')

In [ ]:
datetimeevents = pd.read_csv(datetimeevents_path, compression='gzip')

# 결측치 확인
datetimeevents_missing = datetimeevents.isnull().sum()
print("datetimeevents 데이터 결측치 확인:")
print(datetimeevents_missing[datetimeevents_missing > 0])

# 중복 데이터 확인
datetimeevents_duplicates = datetimeevents.duplicated().sum()
print(f"\ndatetimeevents 데이터 중복된 행의 수: {datetimeevents_duplicates}")

# 중복 데이터가 있는 경우 확인
if datetimeevents_duplicates > 0:
    print("\n중복된 데이터 예시:")
    print(datetimeevents[datetimeevents.duplicated()].head())


datetimeevents 데이터 결측치 확인:
Series([], dtype: int64)

datetimeevents 데이터 중복된 행의 수: 0


### 4. d_items

In [ ]:
# `itemid`와 같은 데이터 항목의 메타데이터를 제공하는 테이블입니다. 각 항목의 이름과 설명이 포함되어 있어, `inputevents`, `outputevents` 등의 다른 테이블에서 사용된 `itemid`의 의미를 알 수 있습니다.

load_and_check_first_chunk(d_items_path)

파일명: /content/drive/MyDrive/AKI/mimic/3.0/icu/d_items.csv.gz
컬럼명: ['itemid', 'label', 'abbreviation', 'linksto', 'category', 'unitname', 'param_type', 'lownormalvalue', 'highnormalvalue']
데이터 샘플:
   itemid                    label        abbreviation         linksto  \
0  220001             Problem List        Problem List     chartevents   
1  220003       ICU Admission date  ICU Admission date  datetimeevents   
2  220045               Heart Rate                  HR     chartevents   
3  220046  Heart rate Alarm - High     HR Alarm - High     chartevents   
4  220047   Heart Rate Alarm - Low      HR Alarm - Low     chartevents   

              category unitname     param_type  lownormalvalue  \
0              General      NaN           Text             NaN   
1                  ADT      NaN  Date and time             NaN   
2  Routine Vital Signs      bpm        Numeric             NaN   
3               Alarms      bpm        Numeric             NaN   
4               Alarms      b

In [ ]:
d_items = pd.read_csv(d_items_path, compression='gzip')
d_items.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4095 entries, 0 to 4094
Data columns (total 9 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   itemid           4095 non-null   int64  
 1   label            4095 non-null   object 
 2   abbreviation     4095 non-null   object 
 3   linksto          4095 non-null   object 
 4   category         4095 non-null   object 
 5   unitname         1123 non-null   object 
 6   param_type       4095 non-null   object 
 7   lownormalvalue   19 non-null     float64
 8   highnormalvalue  22 non-null     float64
dtypes: float64(2), int64(1), object(6)
memory usage: 288.1+ KB


In [ ]:
# 결측치 확인
d_items_missing = d_items.isnull().sum()
print("d_items 데이터 결측치 확인:")
print(d_items_missing[d_items_missing > 0])

# 중복 데이터 확인
d_items_duplicates = d_items.duplicated(subset=['itemid']).sum()
print(f"\nd_items 데이터 중복된 itemid의 수: {d_items_duplicates}")

# 중복 데이터가 있는 경우 확인
if d_items_duplicates > 0:
    print("\n중복된 데이터 예시:")
    print(d_items[d_items.duplicated(subset=['itemid'])].head())


d_items 데이터 결측치 확인:
unitname           2972
lownormalvalue     4076
highnormalvalue    4073
dtype: int64

d_items 데이터 중복된 itemid의 수: 0


### 5. icustays

In [ ]:
# 중환자실 입실 및 퇴실 정보입니다. 각 환자가 중환자실에 입실한 시간, 퇴실한 시간, 입원 이유 등이 포함됩니다.

load_and_check_first_chunk(icustays_path)

파일명: /content/drive/MyDrive/AKI/mimic/3.0/icu/icustays.csv.gz
컬럼명: ['subject_id', 'hadm_id', 'stay_id', 'first_careunit', 'last_careunit', 'intime', 'outtime', 'los']
데이터 샘플:
   subject_id   hadm_id   stay_id                       first_careunit  \
0    10000032  29079034  39553978   Medical Intensive Care Unit (MICU)   
1    10000690  25860671  37081114   Medical Intensive Care Unit (MICU)   
2    10000980  26913865  39765666   Medical Intensive Care Unit (MICU)   
3    10001217  24597018  37067082  Surgical Intensive Care Unit (SICU)   
4    10001217  27703517  34592300  Surgical Intensive Care Unit (SICU)   

                         last_careunit               intime  \
0   Medical Intensive Care Unit (MICU)  2180-07-23 14:00:00   
1   Medical Intensive Care Unit (MICU)  2150-11-02 19:37:00   
2   Medical Intensive Care Unit (MICU)  2189-06-27 08:42:00   
3  Surgical Intensive Care Unit (SICU)  2157-11-20 19:18:02   
4  Surgical Intensive Care Unit (SICU)  2157-12-19 15:42:24   

 

In [ ]:
icustays = pd.read_csv(icustays_path, compression='gzip')

# 결측치 확인
icustays_missing = icustays.isnull().sum()
print("icustays 데이터 결측치 확인:")
print(icustays_missing[icustays_missing > 0])

# 중복 데이터 확인
icustays_duplicates = icustays.duplicated().sum()
print(f"\nicustays 데이터 중복된 행의 수: {icustays_duplicates}")

# 중복 데이터가 있는 경우 확인
if icustays_duplicates > 0:
    print("\n중복된 데이터 예시:")
    print(icustays[icustays.duplicated()].head())

icustays 데이터 결측치 확인:
outtime    14
los        14
dtype: int64

icustays 데이터 중복된 행의 수: 0


### 6. ingredientevents

In [ ]:
# 복합 약물이나 수액 등 여러 성분이 포함된 처치 이벤트의 개별 성분 정보를 기록한 데이터입니다. 예를 들어, 특정 약물 성분의 투입량 등을 기록합니다.

load_and_check_first_chunk(ingredientevents_path)

파일명: /content/drive/MyDrive/AKI/mimic/3.0/icu/ingredientevents.csv.gz
컬럼명: ['subject_id', 'hadm_id', 'stay_id', 'caregiver_id', 'starttime', 'endtime', 'storetime', 'itemid', 'amount', 'amountuom', 'rate', 'rateuom', 'orderid', 'linkorderid', 'statusdescription', 'originalamount', 'originalrate']
데이터 샘플:
   subject_id   hadm_id   stay_id  caregiver_id            starttime  \
0    10000032  29079034  39553978         18704  2180-07-23 17:00:00   
1    10000032  29079034  39553978         18704  2180-07-23 17:00:00   
2    10000032  29079034  39553978         18704  2180-07-23 17:00:00   
3    10000032  29079034  39553978         18704  2180-07-23 17:00:00   
4    10000032  29079034  39553978         18704  2180-07-23 17:33:00   

               endtime            storetime  itemid      amount amountuom  \
0  2180-07-23 17:01:00  2180-07-23 18:56:00  220490  200.000000        mL   
1  2180-07-23 17:01:00  2180-07-23 18:56:00  227075  200.000000        mL   
2  2180-07-23 17:30:00  2180-0

In [ ]:
ingredientevents = pd.read_csv(ingredientevents_path, compression='gzip')

# 결측치 확인
ingredientevents_missing = ingredientevents.isnull().sum()
print("ingredientevents 데이터 결측치 확인:")
print(ingredientevents_missing[ingredientevents_missing > 0])

# 중복 데이터 확인
ingredientevents_duplicates = ingredientevents.duplicated().sum()
print(f"\ningredientevents 데이터 중복된 행의 수: {ingredientevents_duplicates}")

# 중복 데이터가 있는 경우 확인
if ingredientevents_duplicates > 0:
    print("\n중복된 데이터 예시:")
    print(ingredientevents[ingredientevents.duplicated()].head())

ingredientevents 데이터 결측치 확인:
rate       5171995
rateuom    5171995
dtype: int64

ingredientevents 데이터 중복된 행의 수: 0


### 7. inputevents

In [ ]:
# 환자에게 투여된 약물이나 영양제 등의 투입 관련 이벤트를 기록한 데이터입니다. 투입된 물질의 종류, 양, 투입 시간 등이 포함됩니다.

load_and_check_first_chunk(inputevents_path)

파일명: /content/drive/MyDrive/AKI/mimic/3.0/icu/inputevents.csv.gz
컬럼명: ['subject_id', 'hadm_id', 'stay_id', 'caregiver_id', 'starttime', 'endtime', 'storetime', 'itemid', 'amount', 'amountuom', 'rate', 'rateuom', 'orderid', 'linkorderid', 'ordercategoryname', 'secondaryordercategoryname', 'ordercomponenttypedescription', 'ordercategorydescription', 'patientweight', 'totalamount', 'totalamountuom', 'isopenbag', 'continueinnextdept', 'statusdescription', 'originalamount', 'originalrate']
데이터 샘플:
   subject_id   hadm_id   stay_id  caregiver_id            starttime  \
0    10000032  29079034  39553978         18704  2180-07-23 17:00:00   
1    10000032  29079034  39553978         18704  2180-07-23 17:00:00   
2    10000032  29079034  39553978         18704  2180-07-23 17:33:00   
3    10000032  29079034  39553978         18704  2180-07-23 18:56:00   
4    10000032  29079034  39553978         20925  2180-07-23 21:10:00   

               endtime            storetime  itemid      amount amoun

In [ ]:
inputevents = pd.read_csv(inputevents_path, compression='gzip')

# 결측치 확인
inputevents_missing = inputevents.isnull().sum()
print("inputevents 데이터 결측치 확인:")
print(inputevents_missing[inputevents_missing > 0])

# 중복 데이터 확인
inputevents_duplicates = inputevents.duplicated().sum()
print(f"\ninputevents 데이터 중복된 행의 수: {inputevents_duplicates}")

# 중복 데이터가 있는 경우 확인
if inputevents_duplicates > 0:
    print("\n중복된 데이터 예시:")
    print(inputevents[inputevents.duplicated()].head())

inputevents 데이터 결측치 확인:
rate                          4897231
rateuom                       4897231
secondaryordercategoryname    3185024
totalamount                   1521884
totalamountuom                1518590
dtype: int64

inputevents 데이터 중복된 행의 수: 0


### 8. outputevents

In [ ]:
# 환자의 체외로 배출된 물질(소변, 배설물, 배액 등)과 관련된 이벤트를 기록한 데이터입니다. 배출된 물질의 양과 시간 등을 포함합니다.

load_and_check_first_chunk(outputevents_path)

파일명: /content/drive/MyDrive/AKI/mimic/3.0/icu/outputevents.csv.gz
컬럼명: ['subject_id', 'hadm_id', 'stay_id', 'caregiver_id', 'charttime', 'storetime', 'itemid', 'value', 'valueuom']
데이터 샘플:
   subject_id   hadm_id   stay_id  caregiver_id            charttime  \
0    10000032  29079034  39553978         18704  2180-07-23 15:00:00   
1    10000690  25860671  37081114          8787  2150-11-06 08:00:00   
2    10000690  25860671  37081114          8787  2150-11-06 09:00:00   
3    10000690  25860671  37081114          8787  2150-11-06 10:00:00   
4    10000690  25860671  37081114          8787  2150-11-06 11:00:00   

             storetime  itemid  value valueuom  
0  2180-07-23 16:00:00  226560  175.0       mL  
1  2150-11-06 09:08:00  226559   15.0       mL  
2  2150-11-06 09:08:00  226559   15.0       mL  
3  2150-11-06 09:50:00  226559   12.0       mL  
4  2150-11-06 13:42:00  226559   15.0       mL  



In [ ]:
outputevents = pd.read_csv(outputevents_path, compression='gzip')

# 결측치 확인
outputevents_missing = outputevents.isnull().sum()
print("outputevents 데이터 결측치 확인:")
print(outputevents_missing[outputevents_missing > 0])

# 중복 데이터 확인
outputevents_duplicates = outputevents.duplicated().sum()
print(f"\noutputevents 데이터 중복된 행의 수: {outputevents_duplicates}")

# 중복 데이터가 있는 경우 확인
if outputevents_duplicates > 0:
    print("\n중복된 데이터 예시:")
    print(outputevents[outputevents.duplicated()].head())

outputevents 데이터 결측치 확인:
Series([], dtype: int64)

outputevents 데이터 중복된 행의 수: 0


### 9. procedureevents

In [ ]:
# 중환자실에서 이루어진 의료 절차나 수술 관련 이벤트를 기록한 데이터입니다. 여기에는 수술명, 시술 날짜와 시간, 시술 결과 등이 포함될 수 있습니다.

load_and_check_first_chunk(procedureevents_path)

파일명: /content/drive/MyDrive/AKI/mimic/3.0/icu/procedureevents.csv.gz
컬럼명: ['subject_id', 'hadm_id', 'stay_id', 'caregiver_id', 'starttime', 'endtime', 'storetime', 'itemid', 'value', 'valueuom', 'location', 'locationcategory', 'orderid', 'linkorderid', 'ordercategoryname', 'ordercategorydescription', 'patientweight', 'isopenbag', 'continueinnextdept', 'statusdescription', 'originalamount', 'originalrate']
데이터 샘플:
   subject_id   hadm_id   stay_id  caregiver_id            starttime  \
0    10000032  29079034  39553978       18704.0  2180-07-23 14:43:00   
1    10000032  29079034  39553978           NaN  2180-07-23 14:24:00   
2    10000032  29079034  39553978           NaN  2180-07-23 14:24:00   
3    10000690  25860671  37081114       17393.0  2150-11-02 20:00:00   
4    10000690  25860671  37081114       17393.0  2150-11-02 20:38:00   

               endtime                storetime  itemid   value valueuom  ...  \
0  2180-07-23 14:44:00      2180-07-23 14:43:00  225966     1.0      

In [ ]:
procedureevents = pd.read_csv(procedureevents_path, compression='gzip')

# 결측치 확인
procedureevents_missing = procedureevents.isnull().sum()
print("procedureevents 데이터 결측치 확인:")
print(procedureevents_missing[procedureevents_missing > 0])

# 중복 데이터 확인
procedureevents_duplicates = procedureevents.duplicated().sum()
print(f"\nprocedureevents 데이터 중복된 행의 수: {procedureevents_duplicates}")

# 중복 데이터가 있는 경우 확인
if procedureevents_duplicates > 0:
    print("\n중복된 데이터 예시:")
    print(procedureevents[procedureevents.duplicated()].head())

procedureevents 데이터 결측치 확인:
caregiver_id        157542
valueuom            412401
location            631364
locationcategory    631364
dtype: int64

procedureevents 데이터 중복된 행의 수: 0


## 관계 찾기

In [ ]:
# caregiver_id 결측치가 특정 itemid와 관련이 있는지 확인
missing_caregiver = chartevents[chartevents['caregiver_id'].isnull()]
print(missing_caregiver['itemid'].value_counts())

itemid
220635    13
227442    13
227073    12
220602    12
220615    12
220621    12
220645    12
225624    12
225625    12
225677    12
227443    12
220546    10
227457    10
220228    10
220545    10
227465     6
227466     6
227467     6
227429     4
223679     3
225668     3
227445     3
226062     3
226063     3
220274     3
220632     2
220587     2
225690     2
225612     2
220644     2
227464     2
225634     2
225640     1
225641     1
225642     1
225643     1
225639     1
220734     1
227471     1
225698     1
224828     1
223830     1
220235     1
220224     1
225695     1
Name: count, dtype: int64


특정 itemid에서 결측치 발생:
- caregiver_id가 결측인 경우가 특정 itemid에 집중되어 있습니다. 예를 들어, itemid 220635, 227442, 227073 등에서 결측치가 여러 번 발생했습니다.
이는 해당 itemid와 관련된 데이터들이 본래 의료진 정보 없이 기록되었거나, caregiver_id를 기록할 필요가 없는 데이터일 가능성이 있습니다.

결측치 발생 빈도:
- itemid 220635와 227442의 경우 13번의 결측치가 발생했으며, 그 외에도 여러 itemid에서 10번 이상의 결측치가 발생했습니다. 이런 경우, itemid의 의미를 정확히 이해하고, 해당 항목이 의료진 정보와 관계가 있는지를 판단해야 합니다.

In [ ]:
# 결측치가 있는 itemid 확인
missing_itemids = missing_caregiver['itemid'].unique()

# d_items에서 itemid와 label 조회
itemid_info = d_items[d_items['itemid'].isin(missing_itemids)][['itemid', 'label', 'category']]
print(itemid_info)

      itemid                     label category
30    220224      Arterial O2 pressure     Labs
32    220228                Hemoglobin     Labs
33    220235     Arterial CO2 Pressure     Labs
35    220274               PH (Venous)     Labs
136   220545        Hematocrit (serum)     Labs
137   220546                       WBC     Labs
144   220587                       AST     Labs
145   220602          Chloride (serum)     Labs
149   220615        Creatinine (serum)     Labs
150   220621           Glucose (serum)     Labs
152   220632                       LDH     Labs
153   220635                 Magnesium     Labs
155   220644                       ALT     Labs
156   220645            Sodium (serum)     Labs
158   220734             PH (dipstick)     Labs
329   223679        TCO2 (calc) Venous     Labs
382   223830             PH (Arterial)     Labs
885   224828      Arterial Base Excess     Labs
1395  225612        Alkaline Phosphate     Labs
1399  225624                       BUN  